src/media/js/views/late-customizations.js

```diff
@@ -0,0 +1,118 @@
+define('views/late-customizations',
+    ['apps_widget', 'core/cache', 'core/defer', 'core/l10n', 'core/requests',
+     'core/urls', 'core/z'],
+    function(apps_widget, cache, defer, l10n, requests,
+             urls, z) {
+    'use strict';
+
+    var gettext = l10n.gettext;
+    var API_URL = urls.api.url('late-customizations');
+
+    function getFormData() {
+        var $form = $('form.late-customizations');
+        return {
+            carrier: $form.find('[name="carrier"]').val(),
+            region: $form.find('[name="region"]').val(),
+        };
+    }
+
+    function clearForm($form) {
+        $form.removeClass('late-customizations--manage');
+        $form.find('select').removeAttr('disabled');
+        $form.find('.manage-late-customizations').empty();
+    }
+
+    function manageForm($form) {
+        var data = getFormData();
+        cache.flush();
+        requests.get(urls.api.url('late-customizations', [], data))
+                .then(function(response) {
+            Promise.all(response.objects.map(function(setApp) {
```

Comment:
I'll need to polyfill this or figure out how to make this work with `defer.when` but that didn't seem to want to work before.